### ISSUE TITLE
Issue with @types/socket.io with electronize build

### ISSUE THREAD

**Issue by hhfranzo** [ElectronNET/Electron.NET#188]
Hi,

I'm experiencing an issue when I try to build the application with "electronize build /target win"
The build fails with the following message:

"Packaging app for platform win32 x64 using electron v3.0.0
Failed to locate module "@types/socket.io" from "."
        This normally means that either you have deleted this package already somehow (check your ignore settings if using electron-packager).  Or your module installation failed."

Did I miss to install some packages? Note that I already ran "npm install socket.io -g".
 
Thanks.

**Comment by hhfranzo** [comment]
Forgot to say that I'm using Electron.NET 0.0.10...

**Comment by JimmyTheJ** [comment]
I have been working on this repo for the last couple days as I have had a similar problem since upgrading to Dotnet Core 2.2.

You can fork the repo and change the BuildCommand.cs file for the CLI to do npm install without the --production and that might fix it. I plan on putting in a pull request with some fixes since 0.0.10 release and Dotnet Core 2.2 broke some things at some point soon but not sure exactly when.

**Comment by paillave** [comment]
I experienced the same issue.
The following solved it:
`npm install -save @types/socket.io @types/socket.io-client`

**Comment by paillave** [comment]
I succeeded to make it work:

In the logs it is mentionned `Skip npm install, because node_modules directory exists in <folder>\node_modules`

- Go in the folder `<folder>`
- Run `npm install`
- Come back in the root folder of your project and run again `electronize build /target win`

Interpretation of this solution for developers to solve the bug:
In the electron package prepared by electronnet, the folder node_modules is already prefilled with some npm packages. The problem is that this folder doesn't match what is described in the package.json in the same folder. As the packaging (I suppose for better performances) refuses to run `npm install` if the node_modules already exists the final package is attempted to be build with an incomplete set of npm packages, and therefore will fail.

**Comment by paillave** [comment]
I found the reason of the bug (it actually seems to be a bug in ElectronNET).

On a fresh project (or at least, a where the folder `obj/desktop/XXX` doesn't exist already), the build with ElectronNET version 0.0.10 can't possibly work.

Here is the content of `package.json` that will be used to build electron host (the one in the folder `obj\desktop\win`): https://github.com/ElectronNET/Electron.NET/blob/dc3eec37bb1c6caf8c486f2dfeba1861f3d46835/ElectronNET.Host/package.json#L11-L21

The problem is that the automatic batch executes an `npm install --production` like we can see in log and also in the source code of ElectronNET: https://github.com/ElectronNET/Electron.NET/blob/dc3eec37bb1c6caf8c486f2dfeba1861f3d46835/ElectronNET.CLI/Commands/BuildCommand.cs#L94
The problem is that `--production` argument for `npm install` asks npm to ignore what is mentioned in the `devDependencies` node. That is why electron is lost when trying to make the final build of the application as `socket.io` types are missing for typescript compilation.

This `--production` argument has been added by @GregorBiswanger for a big refactoring in the commit 524e6c8. 
I could push a fix, but I can imagine he had a good reason to add this problematic switch as he is involved in this project since the start. It's better he chooses the proper solution for this issue.

**Comment by JimmyTheJ** [comment]
Yeah I ran into this problem too among others recently, so I just forked the repo and fixed it myself for the purposes of the project I'm working on. You can build the CLI .dll and just run a local version of it instead of the globally installed tool.

**Comment by paillave** [comment]
Thanks @robertmuehsig . According to me, this solves the issue. I'll make an attempt from the moment I'm front of my screen.

**Comment by robertmuehsig** [comment]
Thanks - if you hit any other blocking issues just let me know. I try to push a new version to NuGet.org tomorrow. 

Thanks for your patience!

**Comment by paillave** [comment]
Just tested, works great. Thx!

**Comment by hhfranzo** [comment]
It is still not working under Windows. I just re-installed ElectronNET.CLI and updated the ElectronNET.API in the project (actually I tried with different projects), but I get the same message.

Under Linux (Ubuntu 18.04) I cannot say because I have another issue (with an mvc brand new project). I cannot install electron-package, because it doesn't find package.json.

**Comment by Lonka** [comment]
> It is still not working under Windows. I just re-installed ElectronNET.CLI and updated the ElectronNET.API in the project (actually I tried with different projects), but I get the same message.
> 
I updated ElectronNET.CLI (0.0.11) and ElectronNET.API (0.0.11) as same as you, then build the application got the same error message. But I deleted `obj\desktop\win`  and build again, it's working.
